RateLimiter/Validation/RequestValidator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using RateLimiter.ApiRule.Factory;
+using RateLimiter.Model.Enum;
+using RateLimiter.Model.Request;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace RateLimiter.Validation
+{
+    public class RequestValidator
+    {
+        private static readonly Dictionary<string, List<ApiRequest>> _requestLog = new Dictionary<string, List<ApiRequest>>();
+        private readonly IRateLimitedFactory _rateLimitRuleFactory;
+
+        public RequestValidator(IRateLimitedFactory rateLimitRuleFactory)
+        {
+            _rateLimitRuleFactory = rateLimitRuleFactory;
+        }
+        private List<ApiRequest> GetApiRequestsByToken(string token)
+        {
+            _requestLog.TryGetValue(token, out List<ApiRequest> existingRequests);
+
+            return existingRequests;
+        }
+        public bool ValidateRequest(string token, ResourceEnum resourceName)
+        {
+            var resourceRules = _rateLimitRuleFactory.GetRateLimitRulesByResource(resourceName);
+            var tokenRequestLog = GetApiRequestsByToken(token);
```

Comment:
I agree with you , it will bee good if I use resourceName along with token as a key to get user requests for a specific resouce, After that I don't check with specific resourcename 
     in Validate method.